### ISSUE TITLE
`arrayElement` function in clickhouse: What is the run-time complexity?

### ISSUE THREAD

**Issue by saifamp** [ClickHouse/ClickHouse#56318]
We use `arrayElement(<array>, index)` function to pick an item at a specific index from within the array in clickhouse. We were curious about whats the run-time complexity of this function? Is it a constant time operation (i.e O(1) ) or a linear one (i.e. O(n))? 

Let's say, I want to pick the last element in the array, with array length being 100. If I do `arrayElement(<array>, 100)` , will it first scan the 99 ones before returning the 100th element?

**Comment by alexey-milovidov** [comment]
It will do a lookup in constant time, and then create a new column in memory that requires walking through the contents of the original column in memory, which takes linear time.

TLDR: Linear time.
